### PR TITLE
[PLA-8949] Support reading experiment data sources.

### PR DESCRIPTION
**Pull request by anoto-moniz**
Adds support for reading experiment data source references. Adds a property to each branch to allow retrieving the experiment data source contents, if one is associated with it; otherwise, returns `None`. Also allows generating a CSV representing the data contained within the experiment data source.

Currently, the only way to retrieve an experiment data source is through its branch, although a user could of course construct the `ExperimentDataSource` on their own. Presumably we'll want to expose it at other points in the future.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in __version\__.py

### CODE REVIEW

File: src/citrine/informatics/experiment_values.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    value = properties.String('value')
+    typ = properties.String('type', default='CategoricalValue', deserializable=False)
+    
+    def __init__(self, value: str):
+        self.value = value
+
+
+class MixtureExperimentValue(Serializable['MixtureExperimentValue'], ExperimentValue):
+    """A real-valued experiment result."""
```
Comment (anoto-moniz) [inline src/citrine/informatics/experiment_values.py:102]: Missed updating this. Doing so now...

File: src/citrine/informatics/experiment_values.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+    value = properties.String('value')
+    typ = properties.String('type', default='OrganicValue', deserializable=False)
+    
+    def __init__(self, value: str):
+        self.value = value
+
+
+class InorganicExperimentValue(Serializable['InorganicExperimentValue'], ExperimentValue):
+    """A real-valued experiment result."""
```
Comment (anoto-moniz) [inline src/citrine/informatics/experiment_values.py:122]: Missed updating this. Doing so now...

File: src/citrine/informatics/experiment_values.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    """An experiment result mapping ingredients and labels to real values."""
+
+    value = properties.Mapping(properties.String, properties.Float, 'value')
+    typ = properties.String('type', default='MixtureValue', deserializable=False)
+    
+    def __init__(self, value: dict[str, float]):
+        self.value = value
+
+
+class OrganicExperimentValue(Serializable['OrganicExperimentValue'], ExperimentValue):
```
Comment (anoto-moniz) [inline src/citrine/informatics/experiment_values.py:111]: In the backend, `ExperimentValue.OrganicValue` gets mapped to `ChemicalFormula`, just as `OrionVariable.ChemicalFormula` is, which appears in a candidate response. Why is there a difference? _Is_ there a difference? Which is to ask, what should this be called in the SDK?

File: src/citrine/informatics/experiment_values.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    def get_type(cls, data) -> Type[Serializable]:
+        """Return the subtype."""
+        return {
+            "RealValue": RealExperimentValue,
+            "IntegerValue": IntegerExperimentValue,
+            "CategoricalValue": CategoricalExperimentValue,
+            "MixtureValue": MixtureExperimentValue,
+            "OrganicValue": ChemicalFormulaExperimentValue,
+            "InorganicValue": MolecularStructureExperimentValue,
+        }[data["type"]]
```
Comment (kroenlein) [inline src/citrine/informatics/experiment_values.py:34]: `typ` elsewhere is in snake_case.  Have we introduced inconsistency into the exchange?

**Review by genfx999 — DISMISSED** [review]
lgtm

File: src/citrine/informatics/experiment_values.py (hunk 4 above)
Comment (anoto-moniz) [reply src/citrine/informatics/experiment_values.py:34]: The only place I've seen snake-cased type discriminators is data source references ([data_sources.py](https://github.com/CitrineInformatics/citrine-python/blob/main/src/citrine/informatics/data_sources.py)). All the others I'm aware of ([predictors](https://github.com/CitrineInformatics/citrine-python/blob/main/src/citrine/informatics/predictors/predictor.py), [dimensions](https://github.com/CitrineInformatics/citrine-python/blob/main/src/citrine/informatics/dimensions.py), [design spaces](https://github.com/CitrineInformatics/citrine-python/blob/main/src/citrine/informatics/design_spaces/design_space.py), etc) use camel case.

Comment (kroenlein) [reply src/citrine/informatics/experiment_values.py:34]: All of the artifacts on the data side of the platform are in snake_case.
https://github.com/CitrineInformatics/citrine-python/blob/06bd2cb72b4ab04b7ecc9fe258ed4ac3f0a46635/src/citrine/informatics/data_sources.py#L68-L72
Mixed behavior:
https://github.com/CitrineInformatics/citrine-python/blob/06bd2cb72b4ab04b7ecc9fe258ed4ac3f0a46635/src/citrine/informatics/predictors/predictor.py#L24-L27
DataConcepts.py:
https://github.com/CitrineInformatics/citrine-python/blob/06bd2cb72b4ab04b7ecc9fe258ed4ac3f0a46635/src/citrine/resources/data_concepts.py#L74-L77
gemd-python:
https://github.com/CitrineInformatics/gemd-python/blob/1b6c1866aa5844b5a748a8f89ac64535d2776075/gemd/entity/object/material_run.py#L53
Team.py:
https://github.com/CitrineInformatics/citrine-python/blob/06bd2cb72b4ab04b7ecc9fe258ed4ac3f0a46635/src/citrine/resources/team.py#L141-L144

Comment (kroenlein) [reply src/citrine/informatics/experiment_values.py:34]: Also, all the `typ`s from GemTables:
https://github.com/CitrineInformatics/citrine-python/blob/main/src/citrine/gemtables/rows.py
https://github.com/CitrineInformatics/citrine-python/blob/main/src/citrine/gemtables/variables.py
https://github.com/CitrineInformatics/citrine-python/blob/main/src/citrine/gemtables/columns.py

Comment (anoto-moniz) [reply src/citrine/informatics/experiment_values.py:34]: Fair enough. But that's neither here nor there with regards to citrine-python and this PR.

**Review by genfx999 — APPROVED** [review]

**Pull request merged**